### PR TITLE
Storybook improvements

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [x] Documentation Update

## Description

This PR does few things:

1. It implements official addons for switching background color between `var(--body-bg)` and `var(--card-bg)` as these are usually two backgrounds we put our components against. We this should make it easier to test it.
2. It reimplements theme switcher so it's no longer "built into" the story Canvas - instead it utilizes an add-on made specifically for this and uses toolbar as a switcher container. Previous solution was a bit hacky.

That 2nd change required transitioning `_colors.scss` and `dark.scss` to simply be CSS files (instead of SCSS). There are also two side effect - benefits:
1. Making any changes to dark.scss required restarting the storybook which wasn't very efficient while testing. Now we don't need that. This was also an issue in the product itself.
2. It will make it easier to easily switch themes on the fly (when we're ready to do it) in /settings/customization without necessarily refreshing the page. 

It also cleans up few other things in the Storybook.

(I'm fully aware that "Clear background" doesn't work as expected but that's something I think we can all live with for now. AFAIK this is something that the Addon authors (Storybook team actually) are aware of.

## QA Instructions, Screenshots, Recordings

https://user-images.githubusercontent.com/108287/147095209-724511e4-5634-4650-a7cb-30c694453032.mov

## Added/updated tests?

Not need.

## [Forem core team only] How will this change be communicated?

In Forem.team post.

### CODE REVIEW

File: app/javascript/.storybook/manager.js [hunk 1]
```diff
@@ -1,11 +0,0 @@
-import { addons } from '@storybook/addons';
-import { create } from '@storybook/theming/create';
-
-const crayonsTheme = create({
-  base: 'light',
-  brandTitle: '🖍️ Crayons 🖍️',
```
Comment (ludwiczakpawel) [inline app/javascript/.storybook/manager.js:6]: whatevaaaaa ✂️ ✂️ ✂️

File: app/javascript/.storybook/preview.js [hunk 2]
```diff
@@ -86,27 +32,8 @@ addParameters({
   },
 });
 
-export const Foo = () => <Component />;
```
Comment (ludwiczakpawel) [inline app/javascript/.storybook/preview.js:89]: Some legacy stuff that wasn't used anywhere and in fact was throwing errors in console...

File: app/javascript/.storybook/preview.js [hunk 3]
```diff
@@ -86,27 +32,8 @@ addParameters({
...
-Foo.parameters = {
-  previewTabs: { 'storybook/docs/panel': { index: -1 } },
-};
-
 export const parameters = {
   controls: { expanded: true },
-  options: {
-    storySort: {
-      order: [
```
Comment (ludwiczakpawel) [inline app/javascript/.storybook/preview.js:98]: This was not only outdated but also unnecessary.

File: app/javascript/.storybook/main.js [hunk 4]
```diff
@@ -17,6 +17,8 @@ module.exports = {
     '@storybook/addon-links',
     '@storybook/addon-a11y',
     '@storybook/addon-notes/register-panel',
+    '@storybook/addon-backgrounds',
+    '@etchteam/storybook-addon-css-variables-theme',
```
Comment (nickytonline) [inline app/javascript/.storybook/main.js:21]: For folks interested, check out https://github.com/etchteam/storybook-addon-css-variables-theme

File: app/javascript/.storybook/preview.js [hunk 5]
```diff
@@ -2,75 +2,21 @@ import { h } from 'preact';
...
 import '../../assets/stylesheets/minimal.scss';
 import '../../assets/stylesheets/views.scss';
 import '../../assets/stylesheets/crayons.scss';
+import LightTheme from '!!style-loader?injectType=lazyStyleTag!css-loader!../../assets/stylesheets/config/_colors.css';
+import DarkTheme from '!!style-loader?injectType=lazyStyleTag!css-loader!../../assets/stylesheets/themes/dark.css';
 import '../../assets/javascripts/lib/xss';
 import '../../assets/javascripts/utilities/timeAgo';
 import './storybook.scss';
 
-function addStylesheet(theme = '') {
```
Comment (nickytonline) [inline app/javascript/.storybook/preview.js:14]: Glad to see we don't need this custom bit anymore.

**Review by nickytonline — APPROVED** [review]
Thanks for the cleanup @ludwiczakpawel!

File: app/javascript/.storybook/manager.js (hunk 1 above)
Comment (ludwiczakpawel) [reply app/javascript/.storybook/manager.js:6]: Yea, I'll add something in :)

File: app/javascript/.storybook/preview.js [hunk 6]
```diff
@@ -117,4 +44,26 @@ export const parameters = {
   html: {
     root: '#story-content',
   },
```
Comment (aitchiss) [inline app/javascript/.storybook/preview.js:46]: (blocking) - this needs to be removed now - it was added before to make sure the generated HTML code didn't include the theme switcher. Now that we've also removed the `div` with `id="story-content"` the HTML preview is broken, but if we just remove this whole chunk it'll all work fine 👍 
```suggestion
```

**Review by aitchiss — CHANGES_REQUESTED** [review]
This is really cool - love these improvements. I've requested a small change to get the HTML copy/paste working again, but it's just more ✂️

**Review by citizen428 — APPROVED** [review]

**Review by aitchiss — APPROVED** [review]

**Pull request merged**